### PR TITLE
CORE-14016: add metrics for sessions - send and receive

### PR DESCRIPTION
**Pull request by emilybowe**
Screenshot 2023-05-22 at 18 16 22
Add flow session metrics for messages sent and received.

### CODE REVIEW

File: libs/messaging/kafka-message-bus-impl/build.gradle [hunk 1]
```diff
@@ -36,3 +36,4 @@ dependencies {
     testRuntimeOnly "org.apache.logging.log4j:log4j-slf4j-impl:$log4jVersion"
 }
 
+dependencies { implementation 'org.xerial.snappy:snappy-java:1.1.9.1' }
```
Comment (JamesHR3) [inline libs/messaging/kafka-message-bus-impl/build.gradle]: I think the script you've used to run the combined worker has accidentally given you this diff. You may want to stop your local instance and make sure this has been removed, then update this PR.

File: libs/metrics/src/main/kotlin/net/corda/metrics/CordaMetrics.kt [hunk 2]
```diff
@@ -95,15 +95,15 @@ object CordaMetrics {
...
+         * Metric for the time taken to execute the pipeline (excluding any start lag)
          *
          * Number of pipeline events processed can be inferred from the count of events recorded for this metric.
          */
         object FlowEventPipelineExecutionTime : Metric<Timer>("flow.event.pipeline.execution.time", CordaMetrics::timer)
 
 
         /**
-         * Metric for the time taken to execute the flow (excluding any start lag)
+         * Metric for the time taken to execute the fiber (excluding any start lag)
```
Comment (JamesHR3) [inline libs/metrics/src/main/kotlin/net/corda/metrics/CordaMetrics.kt]: Seeing as you're here - this is the time the fiber is running between two suspension points.

File: libs/metrics/src/main/kotlin/net/corda/metrics/CordaMetrics.kt [hunk 3]
```diff
@@ -116,7 +116,7 @@ object CordaMetrics {
         object FlowPipelineExecutionTime : Metric<Timer>("flow.pipeline.execution.time", CordaMetrics::timer)
 
         /**
-         * Metric for the time taken to execute the flow (excluding any start lag)
+         * Metric for the time taken to execute the fiber (excluding any start lag)
```
Comment (JamesHR3) [inline libs/metrics/src/main/kotlin/net/corda/metrics/CordaMetrics.kt]: This is the total time spent executing user code across the execution time of a flow (as opposed to running platform code or waiting for events).

File: libs/metrics/src/main/kotlin/net/corda/metrics/CordaMetrics.kt [hunk 4]
```diff
@@ -95,15 +95,15 @@ object CordaMetrics {
         object FlowEventLagTime : Metric<Timer>("flow.event.lag", CordaMetrics::timer)
 
         /**
-         * Metric for the time taken to execute the flow (excluding any start lag)
+         * Metric for the time taken to execute the pipeline (excluding any start lag)
```
Comment (JamesHR3) [inline libs/metrics/src/main/kotlin/net/corda/metrics/CordaMetrics.kt]: This is the time taken to process a single event in the flow pipeline. There's another one further down (`FlowPipelineExecutionTime`) that also wants updating - this is the total time spent in the pipeline code across the execution time of a flow (as opposed to time spent waiting for events from Kafka).

File: libs/metrics/src/main/kotlin/net/corda/metrics/CordaMetrics.kt [hunk 5]
```diff
@@ -170,6 +170,17 @@ object CordaMetrics {
...
+         * FLOW SESSION METRICS
+         *
+         * The number of messages received by sessions.
+         */
+        object FlowSessionMessagesReceivedCount: Metric<Counter>("flow.session.messages.received.count", Metrics::counter)
+
+        /**
+         * Number of outbound peer-to-peer data messages replayed.
+         */
+        object FlowSessionMessagesSentCount: Metric<Counter>("flow.session.messages.sent.count", Metrics::counter)
```
Comment (JamesHR3) [inline libs/metrics/src/main/kotlin/net/corda/metrics/CordaMetrics.kt]: This is just the number of messages received by sessions - I think you might have copy-pasted the comment here.

**Review by JamesHR3 — COMMENTED** [review]
Mostly a couple of docs issues in the metrics - code looks good.

**Review by JamesHR3 — APPROVED** [review]

**Review by rick-r3 — APPROVED** [review]

**Pull request merged**